### PR TITLE
front: rewrite kp label layer

### LINKED ISSUE #5797: Console error when adding a psl
### What happened?

When toggling the psl checkbox in the speed limit form, this error appears in the console : 

image

### What did you expect to happen?

There should'nt be any error

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to infra editor
2. Open the speed limit form and select a tracksection
3. Toggle the PSL checkbox
4. Open your console

### What operating system, browser and environment are you using?

- **Browser**: Firefox v118
- **OS**: Ubuntu
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

dev bdf9d88

### PR DESCRIPTION
**Pull request by clarani**
closes #5797

The problem was that if a source is of type geojson, it can not receive a source-layer.

In this PR I:
- renamed and modified a bit the configKPLabelLayer function to harmonize it with the other layers + remove the source layer if not needed
- in SpeedSectionEditionLayers, modified the code to have always the 2 sources
- renamed the SNCF_PSL and SNCF_PSL_SIGNS files

### CODE REVIEW

File: front/src/common/Map/Layers/KPLabel.tsx [hunk 1]
```diff
@@ -1,28 +1,26 @@
 import { ExpressionFilterSpecification } from 'maplibre-gl';
-import { LayerProps } from 'react-map-gl/maplibre';
+import { LayerProps, SymbolLayer } from 'react-map-gl/maplibre';
 
-import { Theme } from 'types';
+import { OmitLayer, Theme } from 'types';
 
-interface PlatformProps {
+export default function getKPLabelLayerProps(params: {
```
Comment (SharglutDev) [inline front/src/common/Map/Layers/KPLabel.tsx]: Maybe this file could be only a `.ts` file instead of `.tsx`

**Review by SharglutDev — APPROVED** [review]
Thank you for this fix ! Lgtm and tested :) Left just one suggestion

**Review by SarahBellaha — APPROVED** [review]
Tested, LGTM. Good job ! 💪

File: front/src/common/Map/Layers/KPLabel.tsx (hunk 1 above)
Comment (clarani) [reply front/src/common/Map/Layers/KPLabel.tsx]: Done ! :) I amended my commit since it was a very small change

**Review by Yohh — APPROVED** [review]
:muscle:

**Review by Math-R — APPROVED** [review]

**Pull request merged**